ibft/storage/store.go

```diff
@@ -38,46 +41,151 @@ func init() {
 	}
 }
 
-// ibftStorage struct
+// participantStorage struct
 // instanceType is what separates different iBFT eth2 duty types (attestation, proposal and aggregation)
-type ibftStorage struct {
+type participantStorage struct {
 	prefix         []byte
+	oldPrefix      string // kept back for cleanup
 	db             basedb.Database
 	participantsMu sync.Mutex
 }
 
-// New create new ibft storage
-func New(db basedb.Database, prefix string) qbftstorage.QBFTStore {
-	return &ibftStorage{
-		prefix: []byte(prefix),
-		db:     db,
+// New create new participant store
+func New(db basedb.Database, prefix spectypes.BeaconRole) qbftstorage.ParticipantStore {
+	role := byte(prefix & 0xff)
+	return &participantStorage{
+		prefix:    []byte{role},
+		oldPrefix: prefix.String(),
```

Comment:
the change in prefix has the same meaning as I mentioned above for the database key change.